sdks/python/apache_beam/io/gcp/bigquery.py

```diff
@@ -1702,7 +1702,8 @@ def _get_destination_uri(self, temp_location):
       logging.debug("gcs_location is empty, using temp_location instead")
     else:
       raise ValueError(
-          '{} requires a GCS location to be provided'.format(
+          '{} requires a GCS location to be provided. Neither option'
+          '--gcs_location nor the fallback --temp_location is set.'.format(
```

Comment:
`gcs_location` is not a pipeline option. It's a constructor argument. Can you error out as such?

Reply:
Thanks for the correction, Pablo!

It looks like that Java and Python handles the usage of temp location for bigquery io differently.
Java's `Read` only uses and validates `tempLocation` but `Write` takes `customGcsLocation` and falls back to `tempLocation`.
Python's `Read` takes `gcs_location` and falls back to `temp_location` while `Write` takes `custom_gcs_temp_location` and falls back to `temp_location`.

It seems that Java's `Read` should also allow specifying a `gcsLocation`.

Reply:
hm yeah this was done after the fact. Perhaps file a JIRA to do this, and maybe assign to me. : ) - or Reuven

Reply:
Thanks, created BEAM-10182 for this.